openmls/src/group/public_group/mod.rs

```diff
@@ -141,6 +147,49 @@ impl PublicGroup {
             .full_leaves()
             .try_for_each(|leaf_node| leaf_node.validate_locally())?;
 
+        // For each non-empty parent node and each entry in the node's unmerged_leaves field:
+        treesync
```

Comment:
Shouldn't this whole block be part of `into_verified` that's called above? In any case it might be good to make it its own function.

Reply:
Fair enough. Let's leave it like this for now.